certbot/main.py

```diff
@@ -653,34 +673,26 @@ def obtain_cert(config, plugins, lineage=None):
         raise
     le_client = _init_le_client(config, auth, installer)
 
-    # SHOWTIME: Possibly obtain/renew a cert, and set action to renew | newcert | reinstall
-    if config.csr is None: # the common case
-        domains, certname = _find_domains_or_certname(config, installer)
-        action, _ = _auth_from_available(le_client, config, domains, certname, lineage)
-    else:
-        assert lineage is None, "Did not expect a CSR with a RenewableCert"
-        _csr_obtain_cert(config, le_client)
-        action = "newcert"
+    if config.csr:
+        cert_path, fullchain_path = _csr_obtain_cert(config, le_client)
+        _report_new_cert(config, cert_path, fullchain_path)
```

Comment:
nit: I think it'd be better to put these two calls in `_csr_obtain_cert` since it pushes more work into the smaller function, let's `_csr_obtain_cert` handle all `certonly --csr` specific work, and allows us to stop returning `(None, None)` which will blow up if we ever start reporting new certs from a dry run.

If this change is made, I'd make the above line `return _csr_obtain_cert(config, le_client)`.